libs/ml/include/ml/model/model.hpp

```diff
@@ -167,6 +166,46 @@ void Model<TensorType>::Compile(OptimiserType optimiser_type, ops::LossType loss
     }
   }
 
+  for (auto const &met : metrics)
```

Comment:
since this takes more than a second to figure out what's going on, please add a comment here pointing out that this bit of code bode adds all the necessary metric nodes to the graph and stores their names in 'metrics_' for later access